### DISCUSSION TITLE
Georreferencing chips in a memory-efficient way

### DISCUSSION THREAD

**Discussion by lcoandrade** [torchgeo/torchgeo#1509] [Q&A]
Hi there!
I have a Kaggle [notebook](https://www.kaggle.com/code/luizclaudioandrade/torchgeo-101) that is correctly georreferencing my predictions.

My predictions are georreferenced using Rasterio MemoryFile, but I need to keep my geo chips in memory to be able to use rasterior's merge function:

```
from rasterio.merge import merge

# Merge the chips using Rasterio's merge function
merged, merged_transform = merge(chips_list)
```

I would like to know if there is a way to use a generator to merge my georreferenced memory chips.

The idea is to be memory efficient.

Any ideias?

**Comment by lcoandrade** [comment]
Is there any official solution to make georreferenced predictions available now?

**Reply by adamjstewart** [reply]
Not yet, see https://github.com/microsoft/torchgeo/pull/560#issuecomment-2630671044 for my latest thoughts. There are quite a few issues holding this up:

* https://github.com/microsoft/torchgeo/issues/30
* https://github.com/microsoft/torchgeo/issues/35
* https://github.com/microsoft/torchgeo/pull/411
* https://github.com/microsoft/torchgeo/issues/1407

This will likely become of more importance to my future research projects so I expect to personally work on this in the next year (no promises though).
